crates/wasmtime/src/runtime/vm/instance/allocator/pooling/unix_stack_pool.rs

```diff
@@ -152,11 +158,11 @@ impl StackPool {
         );
 
         // Remove the guard page from the size
-        let stack_size = self.stack_size - self.page_size;
+        let stack_size = self.stack_size.byte_count() - self.page_size.byte_count();
         let bottom_of_stack = top - stack_size;
-        let start_of_stack = bottom_of_stack - self.page_size;
+        let start_of_stack = bottom_of_stack - self.page_size.byte_count();
         assert!(start_of_stack >= base && start_of_stack < (base + len));
-        assert!((start_of_stack - base) % self.stack_size == 0);
+        assert!((start_of_stack - base) % self.stack_size.byte_count() == 0);
```

Comment:
This code and the similar section below would be good to convert over, but I decided to do this in a followup.